app/Http/Controllers/API/Monster/MonsterAPIController.php

```diff
@@ -2,14 +2,28 @@
 
 namespace App\Http\Controllers\API\Monster;
 
-use App\Http\Controllers\API\Base\BaseAPIController;
+use App\Http\Controllers\API\AbstractAPIController;
 use App\Models\Monsters\Monster;
 use App\Http\Resources\MonsterResource;
+use App\Http\Resources\MonsterListResource;
+use App\Http\Requests\ApiListRequest;
 
-class MonsterAPIController extends BaseAPIController
+class MonsterAPIController extends AbstractAPIController
 {
-    public function list() {
-        dd(Monster::get());
+    public function list(ApiListRequest $request) {
+        $validated = $request->validated();
+
+        $limit = $validated['limit'];
+        $offset = $validated['offset'];
+
+        return response()->json(
+            MonsterListResource::collection(
+                Monster
+                    ::skip($offset)
+                    ->take($limit)
+                    ->get()
+            )
+        );
```

Comment:
Could these be part of the `ApiController`? Just something like `$model . 'ListResource'`?

Reply:
Can't find the classes that way.

Reply:
@DanEvans485 I have done something more or less like what you suggested. Thoughts?